### PR TITLE
Ud task

### PR DESCRIPTION
**Pull request by claudiagreen**
Update task method added to Task controller.
Private method for task_params created.
Destroy method added to task controller.
destroy_task method extracted to mode
Tests currently passing.

### CODE REVIEW

File: app/controllers/goals_controller.rb [hunk 1]
```diff
@@ -6,8 +6,28 @@ def index
...
+    current_goal.destroy
+    render nothing: true, status: :no_content
+  end
+
+
+
+  private
+
+    def goal_params
+      params.require(:goal).permit(:name)
```
Comment (ccfz) [inline app/controllers/goals_controller.rb:25]: we just need: name: params[:name]

File: app/controllers/goals_controller.rb [hunk 2]
```diff
@@ -6,8 +6,28 @@ def index
   end
 
   def create
-    Goal.create_goal(name: params[:name])
+    Goal.create_goal(goal_params)
     render json:{}, status: :created
   end
 
+  def update
+    current_goal = Goal.find(params[:id])
```
Comment (ccfz) [inline app/controllers/goals_controller.rb]: we could outsource this and the next line to the model => Goal.update

File: app/controllers/goals_controller.rb [hunk 3]
```diff
@@ -6,8 +6,28 @@ def index
...
+    current_goal.update(goal_params)
+    render json:{}, status: :accepted
+  end
+
+  def destroy
+    current_goal = Goal.find(params[:id])
```
Comment (ccfz) [inline app/controllers/goals_controller.rb]: same was with update we could refactor this to the model with Goal.destroy

File: spec/features/goal_feature_spec.rb [hunk 4]
```diff
@@ -20,14 +20,47 @@
...
+    before :each do
+      FactoryGirl.create_list(:goal, 3)
+    end
+    
+    context 'delete goals' do
+      it 'deletes a goal entry' do
+        delete("/goals/1")
+
+        expect(response.status).to eq 204
+        expect(Goal.count).to eq(2)
```
Comment (ccfz) [inline spec/features/goal_feature_spec.rb]: Maybe we can do the "to have changed by 2" matcher here

**Comment by ccfz** [comment]
done with comments

**Comment by hanfak** [comment]
All comment for `Goal` is done in branch `ud-goal`

**Pull request merged**